### ISSUE TITLE
Client creation (setOption) with nil struct pointer

### ISSUE THREAD

**Issue by nammn** [olivere/elastic#1557]
Please use the following questions as a guideline to help me answer
your issue/question without further inquiry. Thank you.

### Which version of Elastic are you using?

[x] elastic.v7 (for Elasticsearch 7.x)
[ ] elastic.v6 (for Elasticsearch 6.x)
[ ] elastic.v5 (for Elasticsearch 5.x)
[ ] elastic.v3 (for Elasticsearch 2.x)
[ ] elastic.v2 (for Elasticsearch 1.x)

### Please describe the expected behavior

I can call `SetHttpClient` with a nil pointer and with this the `http.DefaultClient` will be set. 

### Please describe the actual behavior

The 7.x. client changed the client set option to use an interface instead in commit: https://github.com/olivere/elastic/commit/282d62ebda7caa7b32506e1f45689c7c264be947

That means that this check: 
https://github.com/olivere/elastic/blob/282d62ebda7caa7b32506e1f45689c7c264be947/client.go#L470-L481

can be `nil` for nil inputs due to the `nil type` and `value check`. Is that wanted? For me it seems to be, in that situation, counterintuitive. 

Here an contrived test with unexpected behaviour: https://go.dev/play/p/vrCAuMcAcZW

### Any steps to reproduce the behavior?

Here an contrived test with unexpected behaviour: https://go.dev/play/p/vrCAuMcAcZW

**Comment by olivere** [comment]
In Go, for an interface to be `nil` both its type and its value need to be `nil`. As you  declared `client` to be of type `*realClient`, the interface will never be `nil`. You can change that by using `client elastic.Doer`. That's just the way Go works, see [here](https://go.dev/doc/faq#nil_error).

**Comment by olivere** [comment]
I'm sorry if I misunderstood. What is the actual change you want to see instead?

**Comment by nammn** [comment]
Oh, let me rephrase instead. I wasn't aiming to say that go is not working the way it should. It does for sure. 

I do think it is more user friendly if we check for the value and not both (value and type) to be empty instead. 
As this can lead to regression updating the client between versions containing linked commit.

In example at some places we needed to add the following in our code to be backwards compatible:
```
	opts := []elastic.ClientOptionFunc{...}
	if cfg.HttpClient == nil {
		elastic.SetHttpClient(nil)
	} else {
		elastic.SetHttpClient(cfg.HttpClient)
	}
```
